pkg/internalapis/cnsvolumeoperationrequest/v1alpha1/cnsvolumeoperationrequest_types.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CnsVolumeOperationRequestSpec defines the desired state of CnsVolumeOperationRequest
+type CnsVolumeOperationRequestSpec struct {
+	// Name represents the name of the instance.
+	// There is no strict naming convention for instances; it is dependent on the caller.
+	Name string `json:"name,omitempty"`
+}
+
+// CnsVolumeOperationRequestStatus defines the observed state of CnsVolumeOperationRequest
+type CnsVolumeOperationRequestStatus struct {
+	// ID represents the VolumeID or SnapshotID of the backend volume or snapshot.
+	// Populated during successful CreateVolume or CreateSnapshot calls.
+	ID string `json:"id,omitempty"`
```

Comment:
ID should not be empty, can we remove omitempty

Reply:
ID is not always populated. It is only populated for CreateVolume and CreateSnapshot APIs

Reply:
Can we have separate fields for volumeId and snapshotId?